server/routes/students.js

```diff
@@ -0,0 +1,62 @@
+const router = require('express').Router();
+const { Student, Test } = require('../db/models');
+
+// GET /api/students
+router.get('/', async (req, res, next) => {
+  try {
+    const students = await Student.findAll({
+      include: { all: true }
+    });
+    res.json(students);
+  } catch (error) {
+    next(error);
+  }
+});
+
+// GET /api/students/:id
+router.get('/:id', async (req, res, next) => {
+  try {
+    const student = await Student.findByPk(req.params.id);
+    if (student) res.json(student);
+    else res.sendStatus(404);
+  } catch (error) {
+    next(error);
+  }
+});
+
+// PUT /api/students/:id
+router.put('/:id', async (req, res, next) => {
+  try {
+    const { firstName, lastName, email } = req.body;
+    const [rowsAffected, affectedRows] = await Student.update(
+      { firstName, lastName, email },
+      {
+        where: {
+          id: req.params.id
+        },
+        returning: true
+      }
+    );
+    const updatedStudent = affectedRows[0];
+    res.json(updatedStudent);
+  } catch (error) {
+    next(error);
+  }
+});
+
+// DELETE /api/students/:id
+router.delete('/:id', async (req, res, next) => {
+  try {
+    const studentDeleted = await Student.destroy({
```

Comment:
Same sentiment here about the class method destroy vs instance - opting for instance

Reply:
Also - Model.destroy returns the number of affected rows, so I'm not sure how much sense it makes to call the variable `studentDeleted` since it's not a boolean. Using the instance.destroy, it returns something different (not too sure) but it might not make sense to even store a variable with the result. Using the instance destroy could be a good opportunity to check if the student even exists, too, before calling destroy on it.

Reply:
`studentDeleted` will either be 0 (falsy) or a number (truthy), but either way, it's been refactored already to use `instance.destroy` instead.